index.php

```diff
@@ -43,27 +43,33 @@
 
 // We are going to need lots of memory and time.
 raise_memory_limit(MEMORY_HUGE);
-set_time_limit(300);
+set_time_limit(600);
 
 $mform = new local_codechecker_form(new moodle_url('/local/codechecker/'));
 $mform->set_data((object)$pageparams);
 if ($data = $mform->get_data()) {
     redirect(new moodle_url('/local/codechecker/', $pageparams));
 }
 
-if ($path) {
-    $fullpath = $CFG->dirroot . '/' . trim($path, '/');
-    if (!is_file($fullpath) && !is_dir($fullpath)) {
-        $fullpath = null;
-    }
-}
-
 $output = $PAGE->get_renderer('local_codechecker');
 
 echo $OUTPUT->header();
 
-if ($path) {
-    if ($fullpath) {
+if ($pathlist) {
+    $paths = preg_split('~[\r\n]+~', $pathlist);
+
+    $ok = true;
+    $fullpaths = [];
+    foreach ($paths as $path) {
+        $fullpath = $CFG->dirroot . '/' . trim($path, '/');
+        if (!is_file($fullpath) && !is_dir($fullpath)) {
+            $fullpaths = [];
```

Comment:
Why an incorrect entry does reset all full-paths? Sure I'm missing something, hence asking.

Reply:
The reason for clearing the array is so that the code underneath (that actually does the checking) does not run. I thought it was better to give an error if a path was invalid, rather than still processing all the other ones. I'll add a comment to explain this & also check the behaviour actually works.

Reply:
(as mentioned in comment below, I actually changed the way this bit works instead, hopefully it makes more sense now.)